spec/controllers/users/registrations_controller_spec.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Users::RegistrationsController, type: :request do
+  context "GET /users/sign_up" do
+    it "properly validates route" do
+      get new_user_registration_path, xhr: true
+      expect(response.status).to eq 200
+    end
+  end
+
+  context "POST /users" do
+    let(:user) { build(:user) }
+    before_count = User.count
+
+    it "properly validates route" do
+      post user_registration_path, xhr: true, params: { user: FactoryBot.attributes_for(:user) }
+      expect(response.status).to eq 200
+      expect(User.count).not_to eq(before_count)
```

Comment:
don't use code like this
`before_count = User.count`
As I said, only `let`(s)

`expect(User.count).not_to eq(before_count)`
this tests that now that count is not equal to previous
but you expect that only one user is created
so you should check exactly what you need
like this

```
expect {
  post user_registration_path, xhr: true, params: { user: FactoryBot.attributes_for(:user) }
}.to change { User.count }.by(1)
expect(response.status).to eq 200
```